### ISSUE TITLE
Formatting does not take override language into account

### ISSUE THREAD

**Issue by MichielDeMey** [i18next/i18next#919]
Currently, the format function does not take the override language into account when formatting the value.

I've created a demo which shows you the issue:
https://runkit.com/59036fcfe589ce001296d666/59036fcfe589ce001296d667

Plain code:
```node.js
const i18next = require('i18next')

i18next.init({
  lng: 'en',
  lngs: [ 'nl', 'en' ],
  interpolation: {
    format: (value, format, lng) => {
      // Language is "en" here, should be "nl"
      console.log('LANGUAGE', lng)

      return value
    }
  },
  resources: {
    en: {
      translation: {
        hello: 'hello {{name,}}'
      }
    },
    nl: {
      translation: {
        hello: 'hallo {{name,}}'
      }
    }
  }
}, () => {
  // Language is "nl" here
  const hello = i18next.t('hello', {
    lng: 'nl',
    name: 'Michiel'
  })

  console.log(hello)
})

```

**Comment by jamuhl** [comment]
ok...https://github.com/i18next/i18next/blob/master/src/Translator.js#L173

changing that to `res = this.interpolator.interpolate(res, data, options.lng || this.language);` should do the trick...

will be fixed asap.

**Comment by jamuhl** [comment]
should be fixed in i18next@8.1.0

**Comment by MichielDeMey** [comment]
Perfect, that fixed it!
Thanks for taking the time to look into this issue.
